client/packages/common/src/hooks/useUrlQuery/useUrlQueryParams.ts

```diff
@@ -70,18 +70,24 @@ export const useUrlQueryParams = ({
 
     const { key: sort, dir } = initialSort;
     updateQuery({ sort, dir: dir === 'desc' ? 'desc' : '' });
-  }, [initialSort]);
-
-  const updateSortQuery = <T extends RecordWithId>(column: Column<T>) => {
-    const currentSort = urlQuery['sort'];
-    const sort = column.key as string;
-    if (sort !== currentSort) {
-      updateQuery({ sort, dir: '', page: '' });
-    } else {
-      const dir = column.sortBy?.direction === 'desc' ? '' : 'desc';
-      updateQuery({ dir });
-    }
-  };
+  }, [initialSort, updateQuery, urlQuery]);
+
+  // Changes sort key or, if the sort key is already selected, toggles the sort direction.
+  const updateSortQuery = useCallback(
+    <T extends RecordWithId>(column: Column<T>) => {
+      const currentSort = urlQuery['sort'];
+      const sort = column.key as string;
+      if (sort !== currentSort) {
+        // change sort key
+        updateQuery({ sort, dir: '', page: '' });
+      } else {
+        // toggle sort direction
+        const dir = column.sortBy?.direction === 'desc' ? '' : 'desc';
+        updateQuery({ dir });
+      }
+    },
```

Comment:
I added some comments here, are they correct?

@mark-prins I think this method should not do the toggling but just update the sort order, i.e. instead of taking a Column it should just take a SortBy. So this hook would really just update the query params for page, filter and sort, but doesn't know about columns. The toggling should IMHO happen in HeaderCell. If you agree I can put this in.

Reply:
yes, comments correct thanks.
The sort is only called from the header cell, which is already typed with the Column: makes sense to push anything column specific down into that component and keep this function clean. 
So yes, if you don't mind - would be good to have that.